src/m365/pp/commands/environment/environment-get.ts

```diff
@@ -0,0 +1,81 @@
+import { Logger } from '../../../../cli/Logger';
+import { AxiosRequestConfig } from 'axios';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import PowerPlatformCommand from '../../../base/PowerPlatformCommand';
+import { Environment } from '../Environment';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  name: string;
+  asAdmin?: boolean;
+}
+
+class PpEnvironmentGetCommand extends PowerPlatformCommand {
+  public get name(): string {
+    return commands.ENVIRONMENT_GET;
+  }
+
+  public get description(): string {
+    return 'Gets information about the specified Power Platform environment';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-n, --name <name>'
+      },
+      {
+        option: '-a, --asAdmin'
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let url: string = '';
+    if (args.options.asAdmin) {
+      url = `${this.resource}/providers/Microsoft.BusinessAppPlatform/scopes/admin/environments`;
+    }
+    else {
+      url = `${this.resource}/providers/Microsoft.BusinessAppPlatform/environments`;
+    }
+
+    const requestOptions: AxiosRequestConfig = {
+      url: `${url}?api-version=2020-10-01&$filter=name eq '${args.options.name}'`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
```

Comment:
observed something really strange. When I query an existing env I get it, and that's ok. But then one I try to get one that does not exist I get the previous one in the output instead of the error 🤔

![image](https://user-images.githubusercontent.com/58668583/194669389-a0905d32-a9d1-4638-9099-116bcb1ac37e.png)

could we have a look into this?

Reply:
Hi @Adam-it 
Thanks for pointing this. The `filter` param doesn't seem to be working with the API endpoint. 
I have changed the implementation to post filter the result.

Reply:
I understand. But is it possible the API returns items with some max count. Like for example we will get only 100 environments as a list and our filter in code will not actually return anything as the environment, we are looking for would be 101 🤔. Probably this case is quite extreme but 😉. But do we maybe know is there some list limit of environments that are returned by the response 🤔? @appieschot?